services/QuillLMS/client/app/bundles/Teacher/components/premium/school_and_district_premium_modal.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+
+import SchoolSelector, { NOT_LISTED, } from '../shared/school_selector'
+import { requestPost, requestPut, } from '../../../../modules/request';
+import { closeIcon, DropdownInput, Spinner, } from '../../../Shared/index'
+
+export const SCHOOL_PURCHASE_STAGE_TWO = 'school_purchase_stage_two'
```

Comment:
I think renaming usages of the concepts stage_one and stage_two with something more descriptive (e.g. 
stage_1 =>  plan_selection_stage, stage_2 => school_selection_stage) would greatly improve readability of the code as well as facilitating refactoring later.

While you and I know what the stages are, another developer (or even myself in a couple months) looking at the code might find the labels stage_1 and stage_2 opaque. 

You can also replace the magic numbers 2 and 1 here
`const [stage, setStage] = React.useState(startAtStageTwo ? 2 : 1)` 

with constants like PLAN_SELECTION_STAGE = 1 and SCHOOL_SELECTION_STAGE = 2.

An example of why this helps with refactoring later is as follows.  Let's say we wanted to add another stage in between stage_1 and stage_2.  We'd first have to go through the error-prone strategy of renaming all stage_2 to stage_3 before adding in the new stage_2.  If we have plan_selection_stage and school_selection_stage, this is no longer a worry.